pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -125,6 +125,9 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: TopologyManager %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	if kc.TopologyManagerScope != kubeletconfig.ContainerTopologyManagerScope && !localFeatureGate.Enabled(features.TopologyManager) {
```

Comment:
shouldn't we have validation that checks for allowable values?

it appears this PR only throws an error at runtime if there is an unsupported value.

Reply:
I'm not sure whether I correctly understood the question. We have two points of validation. I will describe them below:

**1) Passing the _--topology-manager-scope_ flag while _--feature-gates="TopologyManager=false"_ is specified**
---
There are two values:
- `kubeletconfig.ContainerTopologyManagerScope`  <---- default, will be set if flag `--topology-manager-scope` is not specified
- `kubeletconfig.PodTopologyManagerScope`

And this is the check: https://github.com/kubernetes/kubernetes/blob/3aea58766267aaa9440014c7721f56f8446d9f53/pkg/kubelet/apis/config/validation/validation.go#L128-L130

And basically there are three scenarios:
1) scope flag not specified + `--feature-gates="TopologyManage=false"` = no error
2) `--topology-manager-scope=container` + `--feature-gates="TopologyManage=false"` = no error, because we can't differentiate 1) and 2) here, because of default value 
3) `--topology-manager-scope=pod/anything` + `--feature-gates="TopologyManage=false"` = error


**2) Validate for supported and unsupported values**
---
It is done inside Topology Manager, the same as with Topology Manager policies: https://github.com/kubernetes/kubernetes/blob/9022b352ac460ba3280ba135fc5cebd7ac9382a8/pkg/kubelet/cm/topologymanager/topology_manager.go#L150-L161

Reply:
@derekwaynecarr 
> shouldn't we have validation that checks for allowable values?
Is this question for something different than I described above?

Reply:
my comment was that in case number 3 you outlined, it seemed a better experience for the user to validate that the topology-manager-scope was a valid value when we validate the config so the error can be tied to the field in question rather than what this PR appears to do which is validate the policy token later at runtime which means the error is not tied back to the command line flag that was misconfigured.

Reply:
sure, I understand now. I pushed two commits, one for scope validation and one for policy validation, to be consistent.

Reply:
@derekwaynecarr, you can see checks below:
https://github.com/kubernetes/kubernetes/blob/65933f04447f30aab47a4ae18938fc96a5b3bc7f/pkg/kubelet/apis/config/validation/validation.go#L125-L141